components/Plate/plugins/tag.jsx

```diff
@@ -4,7 +4,7 @@ import { createPluginFactory } from "@udecode/plate";
 import { ELEMENT_TAG } from "../../../utils/slate";
 
 // inspired by https://stackoverflow.com/a/60972027
-const tagRegex = /#(\w{1,100})(.*)/
+const tagRegex = /#([\w\-]{1,100})(.*)/
```

Comment:
why {1,100}?

Reply:
Actually, with the second capture group I am not sure what the numbers are even doing. Can you explain?

Reply:
just couldn't imagine a legit use of tags longer than 100 chars, and want to have some limit in there since we store tags in the Garden resource.

Reply:
definitely open to raising the limit, but do want _some_ limit in there

Reply:
What is the `.*` for then?

Reply:
ah yes I had forgotten about that - the normalizer uses that to decide whether to "split" the tag element - see the bit where we look at "extra" and make a decision about splitting

Reply:
probably worth a comment next to the regex!